src/main/java/org/jboss/ejb/client/EJBRootContext.java

```diff
@@ -83,9 +83,15 @@ protected Object lookupNative(final Name name) throws NamingException {
             }
         }
         if (beanName == null) {
+            if (size == 3) {
+                // name is of the form appName/moduleName/distinctName
+                return new RelativeContext(new FastHashtable<>(getEnvironment()), this, SimpleName.of(name));
+            }
```

Comment:
Ah I was interpreting you backwards: I read that we did *not* want to support three segments under any circumstances.  But if this was the legacy behavior then we definitely do need to preserve it.